### DISCUSSION TITLE
How do I install pymupdf in anaconda?

### DISCUSSION THREAD

**Discussion by door-9** [pymupdf/PyMuPDF#1673] [Installation Topics]
I used " conda install -c tc06580 pymupdf " to install pymupdf in anaconda.

But Installation failed with the following instructions.

------------------------------------------------------------------------------------------------------------

Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - pymupdf

Current channels:

  - https://conda.anaconda.org/tc06580/win-64
  - https://conda.anaconda.org/tc06580/noarch
  - https://conda.anaconda.org/conda-forge/win-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/win-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/win-64
  - https://repo.anaconda.com/pkgs/r/noarch
  - https://repo.anaconda.com/pkgs/msys2/win-64
  - https://repo.anaconda.com/pkgs/msys2/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.

------------------------------------------------------------------------------------------------------------

I have confirmed that pymupdf exists in the webpage guided above, but it still does not install.

How do I install pymupdf in anaconda?

**Comment by JorjMcKie** [comment]
PyMuPDF cannot be installed via conda install. You must always use PIP currently. Sorry.

I am not using Anaconda, so need help to bring PyMuPDF through the tedious registration process. Also have no time to dig into all the details. Several people asked me like you did, but nobody cared to jump in yet. So I guess it's not important enough.

I have no idea how PyMuPDF could make it into the list you have been looking at.

**Comment by alephreish** [comment]
Anything that can be installed with pipy can be installed with conda. Just activate the environment, install pipy (`conda install pip`) if it is not installed and do `pip install PyMuPDF` or `pip install PyMuPDF==1.22.5` to pin a version. With yaml environmental file:

```
$ cat pymudpdf.yaml
dependencies:
 - pip
 - pip:
   - PyMuPDF==1.22.5
$ conda env create -n pymudpdf_env -f pymudpdf.yaml
```

conda can obviously be replaced with mamba in both cases.
